inst/include/dust/dust.hpp

```diff
@@ -520,17 +532,22 @@ class Dust {
       }
 
       // We could move this below if wanted but we'd have to rewrite
-      // the re-sort algorithm.
-      if (save_history) {
-        state(filter_state_.history_value_iterator());
+      // the re-sort algorithm; that would be worth doing I think
```

Comment:
Can you expand on this/add an issue? Does 'below' mean after running `resample`?

Reply:
yeah, exactly. Atm we collect the state, then resort - when reconstructing the tree we rely on that order. But we could do both steps at once which reduces the amount of logic in the function but requires rewriting some very painful code